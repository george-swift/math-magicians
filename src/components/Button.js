import PropTypes from 'prop-types';
import keySet from '../data/keySet';

const btnStyle = (value) => {
  if (value === 'AC') return 'clear';
  if (value === '\u002e') return 'decimal';
  return (typeof value === 'number') ? 'number' : 'operator';
};

const Button = ({ name }) => (
  <button
    id={name}
    type="button"
    className={btnStyle(keySet[name])}
    name={name}
    value={keySet[name]}
  >
    {keySet[name]}
  </button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Button;
