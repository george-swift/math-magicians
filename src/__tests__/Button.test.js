import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../components/Button';

describe('Testing button component with passed props', () => {
  const handleClick = jest.fn();

  test('Button renders correctly', () => {
    const tree = renderer
      .create(<Button name="number-5" clickHandler={handleClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Button supports unicode escape sequence', () => {
    const tree = renderer
      .create(<Button name="division" clickHandler={handleClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Calls onClick prop when clicked', () => {
    render(<Button name="percentage" clickHandler={handleClick} />);
    fireEvent.click(screen.getByText(/%/));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
