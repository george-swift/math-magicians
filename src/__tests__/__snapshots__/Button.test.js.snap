// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="number"
  id="number-5"
  name="number-5"
  type="button"
  value={5}
>
  5
</button>
`;

exports[`Button supports unicode escape sequence 1`] = `
<button
  className="operator"
  id="division"
  name="division"
  type="button"
  value="/"
>
  /
</button>
`;
